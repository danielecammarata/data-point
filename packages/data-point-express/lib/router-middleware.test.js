/* eslint-env jest */

const Express = require("express");
const request = require("supertest");
const DataPoint = require("data-point");

const RouterMiddleware = require("./router-middleware");

describe("createRoutes", () => {
  test("It should create routes from options", () => {
    const mid = () => 1;
    const routes = {
      a: {
        path: "/a",
        method: "get",
        middleware: mid
      }
    };
    const app = new Express();
    app.get = jest.fn();
    RouterMiddleware.createRoutes(app, "/api", routes, {});
    expect(app.get).toBeCalledWith("/api/a", mid);
  });
});

describe("setupRoutes", () => {
  test("It should error if routes are missing", () => {
    expect(() => {
      RouterMiddleware.setupRouter();
    }).toThrowErrorMatchingSnapshot();
  });
});

describe("create - datapoint routes", () => {
  let dataPoint;
  beforeAll(() => {
    dataPoint = DataPoint.create({
      entities: {
        "reducer:Test": (value, acc) => ({
          message: `Hello ${acc.locals.query.name}`
        })
      }
    });
  });
  test("It should return 400 if entityId query is missing", done => {
    const app = new Express();
    app.use(
      "/api",
      RouterMiddleware.create(dataPoint, {
        test: {
          priority: 100,
          path: "/test",
          middleware: "reducer:Test"
        }
      })
    );

    request(app)
      .get("/api/test?name=test")
      .expect("Content-Type", /json/)
      .expect(({ body }) => {
        expect(body).toEqual({
          message: "Hello test"
        });
      })
      .expect(200)
      .end(done);
  });
});
