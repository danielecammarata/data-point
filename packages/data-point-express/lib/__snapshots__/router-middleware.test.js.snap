// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setupRoutes It should error if routes are missing 1`] = `"Routes option must be provided"`;
